Build/Chore: cleanup dependencies

m-kolomanski opened the pull request:
## Issue

Closes #

## Description

Change description.

## Definition of Done

Definition of done, preferably copied from the issue.

## How to test

How to test features not covered by unit tests.

## Contributor checklist
- [ ] Code passes lintr checks
- [ ] Code passes all unit tests
- [ ] New logic covered by unit tests
- [ ] New logic is documented
- [ ] Package version is incremented

## Notes to reviewer

Anything that the reviewer should know before tacking the pull request?

Gero1999 reviewed (changes requested):
Hey! maybe I am doing something, when I do `devtools::test()` all runs perfectly. But `devtools::check()` finds problems with the TLG graph function and with `nestcolor` (despite that I don't see in the code any nestcolor call):


## Checking tests
```
.add_label_attribute: no visible binding for global variable
PKNCA_calculate_nca: no visible binding for global variable 'exclude'
apply_labels: no visible binding for global variable 'Dataset'
calculate_summary_stats: no visible binding for global variable 'PPSTRES'
check_app_dependencies: no visible global function definition for 'install.packages'
create_start_impute: no visible binding for global variable 'DOSNO'
export_cdisc: no visible binding for global variable 'end'
l_pkconc : <anonymous>: no visible binding for global variable 'id_list'
____________________________________________________________________________________________________________________________

### Summary (not literal output)
Missing global variable bindings for PPTESTCD, PPSTRESU, PPORRES, PPORRESU, start, end, start_dose, end_dose, exclude, USUBJID, ANALYTE, PCSPEC, DOSNO, PARAM, PCRFTDTM, PCSEQ, PPSTRESN, PPSTRESC, RANGE, is.included.hl, is.excluded.hl, AFRLT, ARRLT, time_dose, EVID, AVAL, Mean, SD, N, SE, CI_lower, CI_upper, SD_min, SD_max, log10_Mean, log10_SD, log10_SE, log10_CI, var_name, Label, digits, zero_str, format_fun, id_list, exclude_half.life, TIME, IX, IX_color, IX_shape, IX_stroke, Spec1_Value, Spec1_Label, Spec1_Units, Spec2_Value, Spec2_Label, Spec2_Units, Ratio, Unit, Ratio_Type, type_interval, lambda.z.time.first, lambda.z.n.points, lambda.z, lambda.z.ix, lambda.z.method, interval_name, interval_name_col, id_plot, view.

Undefined function calls for is, plot_ly, :=, setNames, median, install.packages, log_warn, na.omit, pmap_chr, data.

Suggestions to add imports from methods, stats, and utils to the NAMESPACE file to resolve these issues.
```


```
[ FAIL 9 | WARN 0 | SKIP 0 | PASS 296 ]

══ Failed tests ════════════════════════════════════════════════════════════════
── Error ('test-g_pkconc_ind.R:27:5'): pkcg01: generates valid plot list with default settings ──
<packageNotFoundError/error/condition>
Error in `loadNamespace(x)`: there is no package called 'nestcolor'
Backtrace:
    ▆
 1. ├─aNCA::pkcg01(adpc) at test-g_pkconc_ind.R:27:5
 2. │ └─tern::g_ipp(...)
 3. │   └─base::lapply(...)
 4. │     └─tern (local) FUN(X[[i]], ...)
 5. │       └─tern::h_g_ipp(...)
 6. └─base::loadNamespace(x)
 7.   └─base::withRestarts(stop(cond), retry_loadNamespace = function() NULL)
 8.     └─base (local) withOneRestart(expr, restarts[[1L]])
 9.       └─base (local) doWithOneRestart(return(expr), restart)
── Error ('test-g_pkconc_ind.R:32:5'): pkcg01: generates plot with custom labels ──
<packageNotFoundError/error/condition>
Error in `loadNamespace(x)`: there is no package called 'nestcolor'
Backtrace:
    ▆
 1. ├─aNCA::pkcg01(...) at test-g_pkconc_ind.R:32:5
 2. │ └─tern::g_ipp(...)
 3. │   └─base::lapply(...)
 4. │     └─tern (local) FUN(X[[i]], ...)
 5. │       └─tern::h_g_ipp(...)
 6. └─base::loadNamespace(x)
 7.   └─base::withRestarts(stop(cond), retry_loadNamespace = function() NULL)
 8.     └─base (local) withOneRestart(expr, restarts[[1L]])
 9.       └─base (local) doWithOneRestart(return(expr), restart)
── Error ('test-g_pkconc_ind.R:51:5'): pkcg01: generates plot with custom labels that include special syntax ──
<packageNotFoundError/error/condition>
Error in `loadNamespace(x)`: there is no package called 'nestcolor'
Backtrace:
    ▆
 1. ├─aNCA::pkcg01(...) at test-g_pkconc_ind.R:51:5
 2. │ └─tern::g_ipp(...)
 3. │   └─base::lapply(...)
 4. │     └─tern (local) FUN(X[[i]], ...)
 5. │       └─tern::h_g_ipp(...)
 6. └─base::loadNamespace(x)
 7.   └─base::withRestarts(stop(cond), retry_loadNamespace = function() NULL)
 8.     └─base (local) withOneRestart(expr, restarts[[1L]])
 9.       └─base (local) doWithOneRestart(return(expr), restart)
── Error ('test-g_pkconc_ind.R:70:5'): pkcg01: generates plot list in respect to grouping variables ──
<packageNotFoundError/error/condition>
Error in `loadNamespace(x)`: there is no package called 'nestcolor'
Backtrace:
    ▆
 1. ├─aNCA::pkcg01(adpc, plotgroup_vars = c("ROUTE", "PCSPEC", "PARAM")) at test-g_pkconc_ind.R:70:5
 2. │ └─tern::g_ipp(...)
 3. │   └─base::lapply(...)
 4. │     └─tern (local) FUN(X[[i]], ...)
 5. │       └─tern::h_g_ipp(...)
 6. └─base::loadNamespace(x)
 7.   └─base::withRestarts(stop(cond), retry_loadNamespace = function() NULL)
 8.     └─base (local) withOneRestart(expr, restarts[[1L]])
 9.       └─base (local) doWithOneRestart(return(expr), restart)
── Error ('test-g_pkconc_ind.R:75:5'): pkcg01: generates plots in respect to provided limits ──
<packageNotFoundError/error/condition>
Error in `loadNamespace(x)`: there is no package called 'nestcolor'
Backtrace:
    ▆
 1. ├─aNCA::pkcg01(adpc_single, xmin = 10, xmax = 150, ymin = 1, ymax = 3) at test-g_pkconc_ind.R:75:5
 2. │ └─tern::g_ipp(...)
 3. │   └─base::lapply(...)
 4. │     └─tern (local) FUN(X[[i]], ...)
 5. │       └─tern::h_g_ipp(...)
 6. └─base::loadNamespace(x)
 7.   └─base::withRestarts(stop(cond), retry_loadNamespace = function() NULL)
 8.     └─base (local) withOneRestart(expr, restarts[[1L]])
 9.       └─base (local) doWithOneRestart(return(expr), restart)
── Error ('test-g_pkconc_ind.R:97:5'): g_pkconc_ind_lin: generates plot with linear scale ──
<packageNotFoundError/error/condition>
Error in `loadNamespace(x)`: there is no package called 'nestcolor'
Backtrace:
     ▆
  1. ├─aNCA::g_pkconc_ind_lin(adpc_single) at test-g_pkconc_ind.R:97:5
  2. │ └─aNCA::pkcg01(adpc = data, scale = "LIN", ...)
  3. │   └─tern::g_ipp(...)
  4. │     └─base::lapply(...)
  5. │       └─tern (local) FUN(X[[i]], ...)
  6. │         └─tern::h_g_ipp(...)
  7. └─base::loadNamespace(x)
  8.   └─base::withRestarts(stop(cond), retry_loadNamespace = function() NULL)
  9.     └─base (local) withOneRestart(expr, restarts[[1L]])
 10.       └─base (local) doWithOneRestart(return(expr), restart)
── Error ('test-g_pkconc_ind.R:105:5'): g_pkconc_ind_log: generates plot with log scale ──
<packageNotFoundError/error/condition>
Error in `loadNamespace(x)`: there is no package called 'nestcolor'
Backtrace:
     ▆
  1. ├─aNCA::g_pkconc_ind_log(adpc_single) at test-g_pkconc_ind.R:105:5
  2. │ └─aNCA::pkcg01(adpc = data, scale = "LOG", ...)
  3. │   └─tern::g_ipp(...)
  4. │     └─base::lapply(...)
  5. │       └─tern (local) FUN(X[[i]], ...)
  6. │         └─tern::h_g_ipp(...)
  7. └─base::loadNamespace(x)
  8.   └─base::withRestarts(stop(cond), retry_loadNamespace = function() NULL)
  9.     └─base (local) withOneRestart(expr, restarts[[1L]])
 10.       └─base (local) doWithOneRestart(return(expr), restart)
── Error ('test-general_lineplot.R:25:5'): general_lineplot functions correctly: returns a ggplot object ──
<packageNotFoundError/error/condition>
Error in `loadNamespace(x)`: there is no package called 'nestcolor'
Backtrace:
    ▆
 1. ├─aNCA::general_lineplot(...) at test-general_lineplot.R:25:5
 2. │ └─tern::g_ipp(...)
 3. │   └─tern::h_g_ipp(...)
 4. └─base::loadNamespace(x)
 5.   └─base::withRestarts(stop(cond), retry_loadNamespace = function() NULL)
 6.     └─base (local) withOneRestart(expr, restarts[[1L]])
 7.       └─base (local) doWithOneRestart(return(expr), restart)
── Error ('test-general_lineplot.R:72:5'): general_lineplot functions correctly: can plot with logarithmic scale ──
<packageNotFoundError/error/condition>
Error in `loadNamespace(x)`: there is no package called 'nestcolor'
Backtrace:
    ▆
 1. ├─aNCA::general_lineplot(...) at test-general_lineplot.R:72:5
 2. │ └─tern::g_ipp(...)
 3. │   └─tern::h_g_ipp(...)
 4. └─base::loadNamespace(x)
 5.   └─base::withRestarts(stop(cond), retry_loadNamespace = function() NULL)
 6.     └─base (local) withOneRestart(expr, restarts[[1L]])
 7.       └─base (local) doWithOneRestart(return(expr), restart)

[ FAIL 9 | WARN 0 | SKIP 0 | PASS 296 ]
Error: Test failures
Execution halted

1 error ✖ | 0 warnings ✔ | 2 notes ✖

```

m-kolomanski commented:
Hey, `tern` imports and uses `nestcolor` and that is what is failing. The check runs in separate process, can it be possible that his process does not have an access to that package installed on your machine? Do you use any environment managing tools like `conda` or `renv` that could interfere with the libraries check process has access to? If so, try disabling the environment tools and install `nestcolor` then run the check again. Additional question, what OS are you using?

rabiibouhestine commented on R/intervals_helpers.R:
```diff
@@ -247,7 +247,7 @@ interval_remove_impute.data.frame <- function(data,
     target_params <- param_cols
   } else {
-    checkmate::assert_subset(target_params, choices = all_param_options)
+    stopifnot(all(target_params %in% all_param_options))
```
same comment for the error messaging here.

rabiibouhestine reviewed (changes requested):
nice work! please see my comments.

m-kolomanski commented:
@rabiibouhestine I have implemented an internal function that asserts the subset with more verbose error, PTAL!

m-kolomanski commented:
@Gero1999 The issue appeared in the CI pipeline, I noticed I have removed `nestcolor` completely from the `DESCRIPTION` file, but it should be in the Suggests category. I have added it back and this fixed the CI - please check if this is also the case for your machine.

rabiibouhestine reviewed (approved):
LGTM

Gero1999 commented:
> Do you use any environment managing tools like `conda` or `renv` that could interfere with the libraries check process has access to? If so, try disabling the environment tools and install `nestcolor` then run the check again. Additional question, what OS are you using?

In principle I am not. I used Windows to run it, I will check again

Gero1999 reviewed (commented):
Thanks for the changes, the app keeps working perfectly and the code review also makes sense to me. I just want to point out that I still have 2 notes, not sure if expected/relevant @m-kolomanski:

```
❯ checking for future file timestamps ... NOTE
  unable to verify current time

❯ checking R code for possible problems ... [17s] NOTE
  .add_label_attribute: no visible binding for global variable 'PPTESTCD'
  .add_label_attribute: no visible binding for global variable 'PPSTRESU'
...
Undefined global functions or variables:
    . := AFRLT AFRLT.dose ANALYTE ARRLT AVAL CI_lower CI_upper
    Count.missing Count.total DOSEA DOSNO Dataset EVID Geocv Geomean IX
    IX_color IX_shape IX_stroke Label Max Mean Median Min N NRRLT PARAM
    PCRFTDTM PCSEQ PCSPEC PPORRES PPORRESU PPSTRES PPSTRESC PPSTRESN
    PPSTRESU PPTESTCD PPTESTCD_unit RANGE Ratio Ratio_Type SD SD_max
    SD_min SE STUDYID Spec1_Label Spec1_Units Spec1_Value Spec2_Label
    Spec2_Units Spec2_Value TIME USUBJID Unit Value data digits dose_time
    end end_dose exclude exclude_half.life format_fun id_list id_plot
    install.packages interval_name interval_name_col interval_next
    interval_prev is is.excluded.hl is.included.hl lambda.z lambda.z.ix
    lambda.z.method lambda.z.n.points lambda.z.time.first log10_CI
    log10_Mean log10_SD log10_SE log_warn median na.omit next_dose
    next_nom_dose nom_dose_time nom_interval_next nom_interval_prev
    plot_ly pmap_chr setNames start start_dose time_dose type_interval
    var_name view zero_str
  Consider adding
    importFrom("methods", "is")
    importFrom("stats", "end", "median", "na.omit", "setNames", "start")
    importFrom("utils", "data", "install.packages")
  to your NAMESPACE file (and ensure that your DESCRIPTION Imports field
  contains 'methods').
```

m-kolomanski commented:
@Gero1999 Yes, I am aware, with the timestamp - this relies on external API that is now down, we cannot do anything about that note. It does not appear in the CI. For the bindings, I would like to dive into it in its own PR if possible.

Gero1999 reviewed (approved):
lgtm!
